Introduce withoutOnConflict flag to sqlite-persister, add type='view' to FROM_PRAGMA_TABLE statement

bndkt opened the pull request:
## Summary

This is the first PR in pursuit of #130. I decided to split up the work in the 1) prerequisites (introducing the ability to read VIEWS and the option to write changes using INSERT OR REPLACE instead of using ON CONFLICT) and 2) the actual PowerSync persister to make this easier to digest. The whole PowerSync adapter is implemented in my fork: https://github.com/bndkt/tinybase/tree/powersync

## How did you test this change?

I made sure all existing tests still succeed. I had to adapt the tests that compare expected to actual SQL statements though, as this change changes some SQL statements (adding views).

jamesgpearce commented on src/persisters/sqlite/commands.ts:
```diff
@@ -52,7 +53,7 @@ export const getCommandFunctions = (
     deleteEmptyColumns: boolean,
     deleteEmptyTable: boolean,
-    partial?: boolean,
+    partial?: boolean
```
Seems like a prettier problem. Are you picking up the project config?

bndkt replied:
Yes that was a prettier problem. I've switched a project of mine to Biome and had disabled the Prettier extension in VS Code.

jamesgpearce commented on src/persisters/sqlite/commands.ts:
```diff
@@ -43,6 +43,7 @@ export const getCommandFunctions = (
   cmd: Cmd,
   managedTableNames: string[],
   onIgnoredError: ((error: any) => void) | undefined,
+  withoutOnConflict = false,
```
I know it's internal so it's not critical, but this feels like a double negative. I might have chosen `useOnConflict` defaulting to `true`... but I'm fine with this if it's too much to change.

bndkt replied:
Agree, changed it!

jamesgpearce commented on src/persisters/sqlite/commands.ts:
```diff
@@ -71,7 +72,8 @@ export const getCommandFunctions = (
                 'list ' +
-                `WHERE schema='main'AND type='table'AND name IN(` +
+                `WHERE schema='main'AND(type='table'OR type='view')` +
+                'AND name IN(' +
```
LOVE that you picked up on and continued my SQL whitespace OCD :)

jamesgpearce reviewed (commented):
Overall, AMAZING work. Thank you so much. Couple of comments inline.

bndkt commented:
Thanks for the great feedback @jamesgpearce! I've implemented the changes you suggested. While flipping the boolean logic, I also noticed that I defined defaults multiple times. I've now changed this to have the default only at the final function where the flag is used, to make it more DRY. Hope you agree with that.

bndkt commented:
Awesome! My suggestion would be to merge this, all the rest will be purely additional (e.g. new files for the new persister and entries in the build files), I don't anticipate any other changes to existing code. Maybe I'll even get the PR ready before 5.0, the code is done and I "only" have to write commends/docs.

jamesgpearce commented:
I take "only" very seriously on this project 😉

jamesgpearce commented:
You missed a couple of renames but I just fixed them anyway while I was rebasing. Good to go!
